File: EuclidsAlgorithm/java/src/euclid/Euclid.java
package euclid;

public class Euclid {
	
	/**
	 * A function for finding the greatest common denominator between
	 * two inputs.
	 *
	 * @return
	 * Returns the Greatest Common Divisor or -1 if the input is invalid
	 * (invalid inputs are any value less than 0). This function returns
	 * 0 if either input is equal to 0.
	 */
	public static int GCD(int input1, int input2) {
		if(input1==0||input2==0) {return 0;}
		if(input1<0||input2<0) {return -1;}

		int remainder;
		while((remainder=input1%input2)!=0) {
			input1 = input2;
			input2 = remainder;
		} 

		return input2;
	}
}

File: CountEuclidsAlgorithm/go/euclid.go
/*
	Package euclid allows for finding the greatest common denominator
	between two numbers.
*/
package main

// Function euclid finds the greates common denominator between two unsigned
// If either value is 0 the result will be 0.
func euclid(input1,input2 uint) (uint,int) {
	var remainder uint
	count:=0
	if(input1==0||input2==0) {return 0,count}
	for {
		count++
		remainder = input1%input2
		if(remainder==0) {break}
		input1 = input2
		input2 = remainder
	}
	return input2,count
}
